assignment1/q2.cpp

```diff
@@ -0,0 +1,33 @@
+struct node{
+    int val;
+    node* next;
+};
+
+node *kth_element(int k, node *head) {
+
+    //number of element on which starts 'k-th to end' fragment
+    int n = length(head) - k;
+    //not sure what to do when n <= 0
+    //return what function found or do sth that shows there was an 'error'?
+
+    int i = 1;  //linked list element no
+    node *ptr = new node;
+    ptr = head;
+    while(i < n && ptr != NULL) {
+        i++;
+        ptr->next;
```

Comment:
This line does nothing.
Did you mean: `ptr = ptr->next;` ?

It is always recommended to try to run your code on several examples, so you can find such bugs :)